src/main/kotlin/app/revanced/patches/grindr/unlimited/fingerprints/IsNoXtraUpsellFingerprint.kt

```diff
@@ -0,0 +1,27 @@
+package app.revanced.patches.grindr.unlimited.fingerprints
+
+import app.revanced.patcher.extensions.or
+import app.revanced.patcher.fingerprint.method.annotation.FuzzyPatternScanMethod
+import app.revanced.patcher.fingerprint.method.impl.MethodFingerprint
+import com.android.tools.smali.dexlib2.AccessFlags
+import com.android.tools.smali.dexlib2.Opcode
+
+object IsNoXtraUpsellFingerprint : MethodFingerprint(
+    "Z",
+    accessFlags = AccessFlags.PUBLIC.value,
+    opcodes = listOf(
+        Opcode.INVOKE_VIRTUAL,
+        Opcode.MOVE_RESULT,
+        Opcode.IF_NEZ,
+        Opcode.INVOKE_VIRTUAL,
+        Opcode.MOVE_RESULT,
+        Opcode.IF_NEZ,
+        Opcode.CONST_4,
+        Opcode.GOTO,
+        Opcode.CONST_4,
+        Opcode.RETURN
+    ),
+    customFingerprint = { methodDef, _ ->
+        methodDef.name.contains("h")
```

Comment:
This will not work across versions, you should not directly check for obfuscated names.